x-pack/plugins/triggers_actions_ui/public/application/sections/alert_form/alert_has_changed.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import deepEqual from 'fast-deep-equal';
+import { pick } from 'lodash';
+import { AlertTypeParams } from '../../../types';
+import { InitialAlert } from './alert_reducer';
+
+const DEEP_COMPARE_FIELDS = ['tags', 'schedule', 'actions', 'notifyWhen'];
+
+function getNonNullCompareFields(alert: InitialAlert) {
+  const { name, alertTypeId, throttle } = alert;
+  return {
+    ...(!!(name && name.length > 0) ? { name } : {}),
+    ...(!!(alertTypeId && alertTypeId.length > 0) ? { alertTypeId } : {}),
+    ...(!!(throttle && throttle.length > 0) ? { throttle } : {}),
+  };
+}
+
+export function alertHasChanged(a: InitialAlert, b: InitialAlert, compareParams: boolean) {
```

Comment:
**nit:** it would be clearer to me if the function / file was called `hasAlertChanged` knowing `has*` functions return booleans.

Reply:
Updated to `hasAlertChanged`. I named it `alertHasChanged` because I was using it with an `if` statement, so it reads `if (alertHasChanged(...))` but if other similar functions use the `has*` structure, it makes sense to update for consistency.